выполнил курсовую

AHTI6IOTIK opened the pull request.

Commit:
выполнил курсовую

dmitry-bizin commented on src/Employee.java:
```diff
@@ -0,0 +1,55 @@
+public class Employee {
+
+    private static int counter;
```
для наглядности лучше задать начально значение, чтобы были понятно, откуда стартуют id

AHTI6IOTIK replied:
поправил

dmitry-bizin commented on src/Employee.java:
```diff
@@ -0,0 +1,55 @@
+    private double salary;
+
+    public Employee(String fullName, Department department, double salary) {
+        this.id = ++Employee.counter;
```
здесь просто можно `++counter`, без имени класса

AHTI6IOTIK replied:
поправил

dmitry-bizin commented on src/EmployeeBook.java:
```diff
@@ -0,0 +1,296 @@
+    public Employee getEmployeeWithMaximumSalary() {
+        Employee maxSalaryEmployee = this.employees[0];
+        for (int i = 0; i < this.size; i++) {
+            if (maxSalaryEmployee.getSalary() < this.employees[i].getSalary()) {
```
вот если после создания книги я дёрну этот метод, то здесь упадёт NPE, потому что под индексом 0 будет null

AHTI6IOTIK replied:
Цикл по идее не должен запуститься тк инициализации не было и this.size = 0

dmitry-bizin replied:
а, точно, сорри)

dmitry-bizin commented on src/EmployeeBook.java:
```diff
@@ -0,0 +1,296 @@
+    }
+
+    public Employee[] getEmployeesByDepartment(Integer department) {
+        ArrayList<Employee> employeesDepartment = new ArrayList<>();
```
слева List лучше использовать

AHTI6IOTIK replied:
перелинковал на интерфейс

dmitry-bizin commented on src/EmployeeBook.java:
```diff
@@ -0,0 +1,296 @@
+    }
+
+    public Employee[] getEmployeesWithLessSalaryByNumber(double number) {
+        ArrayList<Employee> employees = new ArrayList<>();
```
слева List

AHTI6IOTIK replied:
перелинковал на интерфейс

dmitry-bizin commented on src/EmployeeBook.java:
```diff
@@ -0,0 +1,296 @@
+    }
+
+    public Employee[] getEmployeesWithGreaterThanSalaryByNumber(double number) {
+        ArrayList<Employee> employees = new ArrayList<>();
```
слева List

AHTI6IOTIK replied:
перелинковал на интерфейс

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,115 @@
+    private static final int DEPARTMENT_SIZE = 5;
+    private static final int EMPLOYEES_SIZE = 10;
+
+    private static final EmployeeBook employeesBook = new EmployeeBook();
```
поля static final принято называть в верхнем регистре, как у тебя выше 2 поля

AHTI6IOTIK replied:
Поправил имя константы

Commit:
Внес правки по ревью
- Поправил имя константы
- перевел объявление ArrayList на обстракцию
- Дал значение по умолчанию счетчику, поменял обращение к нему

Commit:
Внес правки по ревью

- Поправил имя константы
- перевел объявление ArrayList на обстракцию
- Дал значение по умолчанию счетчику, поменял обращение к нему

Commit:
Поправил ошибку NPE

Commit:
Merge remote-tracking branch 'origin/coursework' into coursework

# Conflicts:
#	src/Main.java
